lib/std/enums.zig

```diff
@@ -1505,3 +1505,70 @@ test values {
     };
     try testing.expectEqualSlices(E, &.{ .X, .Y, .Z }, values(E));
 }
+
+/// Returns an enum containing the combined elements of the given enum types.
+/// The values are incremental integers starting at 0.
+fn CombinedEnums(comptime enum_types: anytype) type {
+    return CombinedEnumsTagged(null, enum_types);
+}
+
+/// Returns an enum with the given tag type containing the combined elements
+/// of the given enum types.
+/// The values are incremental integers starting at 0.
+fn CombinedEnumsTagged(comptime tag: ?type, comptime enum_types: anytype) type {
+    var fieldCount = 0;
+    inline for (enum_types) |enum_type| {
+        fieldCount += std.meta.fields(enum_type).len;
```

Comment:
```suggestion
        fieldCount += @typeInfo(enum_type).@"enum".fields.len;
```

Reply:
it looks like its meant to work on tagged unions as well

Reply:
i got that, my suggestion here was disagreeing. putting unions in and getting an enum out would lead to very confusing behavior to users imo

Reply:
unless there was a strong motivating use case rather than this being suggested for std out of the blue